src/Math-Core-Distribution/PMNormalDistribution.class.st

```diff
@@ -41,20 +41,19 @@ PMNormalDistribution class >> new: aNumber1 sigma: aNumber2 [
 { #category : #information }
 PMNormalDistribution class >> random [
 	"Answer a random number distributed according to a (0,1) normal distribution."
+
 	| v1 v2 w y |
-	NextRandom isNil
-		ifTrue: [ [ v1 := Number random * 2 - 1.
-					v2 := Number random * 2 - 1.
-					w := v1 squared + v2 squared.
-					w > 1 ] whileTrue: [].
-				  y := ( ( w ln * 2 negated) / w) sqrt.
-				v1 := y * v1.
-				NextRandom := y * v2.
-				]
-		ifFalse:[ v1 :=NextRandom.
-				  NextRandom := nil.
-				].
-	^v1
+	NextRandom
+		ifNil: [ [ v1 := Number random * 2 - 1.
```

Comment:
This part is a little hard for me to follow to be honest.  If you could factor out this block to an intention revealing method to begin which, it would be very helpful.

Reply:
I have no idea what the block is doing ...

Reply:
What? It's well known Box-Muller algorithm for transforming a uniform distribution into a normal distribution.
See https://en.wikipedia.org/wiki/Box%E2%80%93Muller_transform

We need a pair (v1,v2) inside the unit circle.
If it's not, we have to redraw another pair.
Just replace `whileTrue: [ ].` with `whileTrue.`

Reply:
I replace ``whileTrue:[]`` with ``whileTrue``. I add a comment for the moment to say that we are using the Box-Muller transform. Do you see a better way to be more intention revealing @hemalvarambhia ?

Reply:
You could extract act boxMullerAlgorithm method